staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -137,86 +169,99 @@ func NewCmdAnnotate(parent string, f cmdutil.Factory, ioStreams genericclioption
+	cmd.Flags().BoolVar(&flags.Local, "local", flags.Local, "If true, annotation will NOT contact api-server but run locally.")
+	cmd.Flags().StringVar(&flags.FieldSelector, "field-selector", flags.FieldSelector, "Selector (field query) to filter on, supports '=', '==', and '!='.(e.g. --field-selector key1=value1,key2=value2). The server only supports a limited number of field queries per type.")
+	cmd.Flags().BoolVar(&flags.All, "all", flags.All, "Select all resources, in the namespace of the specified resource types.")
+	cmd.Flags().BoolVarP(&flags.AllNamespaces, "all-namespaces", "A", flags.AllNamespaces, "If true, check the specified action in all namespaces.")
+	cmd.Flags().StringVar(&flags.ResourceVersion, "resource-version", flags.ResourceVersion, i18n.T("If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource."))
+}
+
+// ToOptions converts from CLI inputs to runtime inputs.
+func (flags *AnnotateFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, args []string) (*AnnotateOptions, error) {
+	options := &AnnotateOptions{
+		AllNamespaces: flags.AllNamespaces,
+		IOStreams:     flags.IOStreams,
+	}
 	var err error
 
-	o.RecordFlags.Complete(cmd)
-	o.Recorder, err = o.RecordFlags.ToRecorder()
+	flags.RecordFlags.Complete(cmd)
+	options.Recorder, err = flags.RecordFlags.ToRecorder()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.outputFormat = cmdutil.GetFlagString(cmd, "output")
-	o.dryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	options.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	dynamicClient, err := f.DynamicClient()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.dryRunVerifier = resource.NewQueryParamVerifier(dynamicClient, f.OpenAPIGetter(), resource.QueryParamDryRun)
 
-	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.dryRunStrategy)
-	printer, err := o.PrintFlags.ToPrinter()
+	options.dryRunVerifier = resource.NewQueryParamVerifier(dynamicClient, f.OpenAPIGetter(), resource.QueryParamDryRun)
+
+	cmdutil.PrintFlagsWithDryRunStrategy(flags.PrintFlags, flags.DryRunStrategy)
+	printer, err := flags.PrintFlags.ToPrinter()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.PrintObj = func(obj runtime.Object, out io.Writer) error {
+	options.PrintObj = func(obj runtime.Object, out io.Writer) error {
 		return printer.PrintObj(obj, out)
 	}
 
-	if o.list && len(o.outputFormat) > 0 {
-		return fmt.Errorf("--list and --output may not be specified together")
+	//is there even a --output or --list
+	if flags.List && len(flags.OutputFormat) > 0 {
+		return nil, fmt.Errorf("--list and --output may not be specified together")
 	}
 
-	o.namespace, o.enforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	options.Namespace, options.enforceNamespace, err = flags.RESTClientGetter.ToRawKubeConfigLoader().Namespace()
```

Comment:
We are using `flags.RestClientGetter` in here but we are also passing same object `cmdutil.Factory` as parameter?

Reply:
Why we are exporting `options.Namespace`?

Reply:
I didn't know better :), Fixed both.